core/src/verified_vote_packets.rs

```diff
@@ -680,6 +700,110 @@ mod tests {
         );
     }
 
+    fn send_vote_state_update_and_process(
+        s: &Sender<Vec<VerifiedVoteMetadata>>,
+        r: &Receiver<Vec<VerifiedVoteMetadata>>,
+        vote: VoteStateUpdate,
+        vote_account_key: Pubkey,
+        feature_set: Option<Arc<FeatureSet>>,
+        verified_vote_packets: &mut VerifiedVotePackets,
+    ) -> GossipVote {
+        s.send(vec![VerifiedVoteMetadata {
+            vote_account_key,
+            vote: VoteTransaction::from(vote),
+            packet_batch: PacketBatch::default(),
+            signature: Signature::new(&[1u8; 64]),
+        }])
+        .unwrap();
+        verified_vote_packets
+            .receive_and_process_vote_packets(r, true, feature_set)
+            .unwrap();
+        match verified_vote_packets.0.get(&vote_account_key).unwrap() {
+            SingleValidatorVotes::FullTowerVote(gossip_vote) => gossip_vote.clone(),
+            _ => panic!("Received incremental vote"),
+        }
+    }
+
+    #[test]
+    fn test_latest_vote_tie_break_with_feature() {
+        let (s, r) = unbounded();
+        let vote_account_key = solana_sdk::pubkey::new_rand();
+
+        // Send identical vote state updates with different hashes / ts
+        let mut vote = VoteStateUpdate::from(vec![(2, 4), (4, 3), (6, 2), (7, 1)]);
+        vote.timestamp = Some(5);
+
+        let mut vote_later_ts = vote.clone();
+        vote_later_ts.timestamp = Some(6);
+
+        let mut vote_earlier_ts = vote.clone();
+        vote_earlier_ts.timestamp = Some(4);
+
+        let mut vote_no_ts = vote.clone();
+        vote_no_ts.timestamp = None;
+
+        let mut verified_vote_packets = VerifiedVotePackets(HashMap::new());
+        let mut feature_set = FeatureSet::default();
+        feature_set.activate(&allow_votes_to_directly_update_vote_state::id(), 0);
+        let feature_set = Some(Arc::new(feature_set));
+
+        // Original vote
+        let GossipVote {
+            slot, timestamp, ..
+        } = send_vote_state_update_and_process(
+            &s,
+            &r,
+            vote,
+            vote_account_key,
+            feature_set.clone(),
+            &mut verified_vote_packets,
+        );
+        assert_eq!(slot, 7);
+        assert_eq!(timestamp, Some(5));
```

Comment:
consider using the values directly out of the expected `VoteStateUpdate` for the rhs here instead of hard coding repeats